tao_compiler/mlir/disc/disc_compiler.cc

```diff
@@ -753,34 +757,106 @@ LogicalResult LowerHLOToSharedLibrary(ModuleOp m,
 namespace tensorflow {
 
 Status ConvertTF2MlirHlo(mlir::ModuleOp module_op) {
-  mlir::PassManager tf2xla(module_op.getContext());
+  mlir::PassManager pm(module_op.getContext());
 
-  tf2xla.getContext()->disableMultithreading();
+  pm.getContext()->disableMultithreading();
   auto printingFlags = mlir::OpPrintingFlags();
   printingFlags.elideLargeElementsAttrs(16);
-  tf2xla.enableIRPrinting(
+  pm.enableIRPrinting(
       /*shouldPrintBeforePass=*/nullptr,
       /*shouldPrintAfterPass=*/
       [](mlir::Pass* pass, mlir::Operation*) { return VLOG_IS_ON(1); },
       /*printModuleScope=*/false,
       /*printAfterOnlyOnChange=*/true,
       /*printAfterOnlyOnFailure*/ false, llvm::dbgs(), printingFlags);
+  bool prefer_tf2xla = false;
+  llvm::StringRef device_type = "XLA_CPU_JIT";
 
-  tf2xla.addPass(mlir::disc_ral::createReviseArgsForStaticRankPass());
+  // Replace const arguments to ConstOp and update argument type if it is a
+  // fixed-shaped input
+  pm.addPass(mlir::disc_ral::createReviseArgsForStaticRankPass());
 
-  mlir::SmallVector<std::unique_ptr<mlir::Pass>, 1> lower_passes;
-  lower_passes.emplace_back(mlir::disc_ral::createDiscLowerTfPass());
-  CreateConvertMlirToXlaHloPipeline(
-      tf2xla, "XLA_CPU_JIT",
-      /*prefer_tf2xla=*/false,
-      /*custom_legalization_passes=*/lower_passes);
+  // Note that the region-based control-flow produced here still contains
+  // function call ops which get inlined by the subsequent inliner pass.
+  pm.addPass(mlir::TF::CreateTFFunctionalControlFlowToRegions());
```

Comment:
Are these added passes related to the support of quantize/dequantize ops?

Reply:
Before this PR, we directly call `CreateConvertMlirToXlaHloPipeline ` to get a pre-defined set of passes to do tf2mhlo conversion. The pre-defined pass pipeline is not able to support customize quantized op lowering, thus we have to build our own pass pipeline.

Reply:
Seems like it's the InlinerPass in the CreateConvertMlirToXlaHloPipeline that breaks the lowering of (de)quantize op?

Reply:
Actually it's `CreateLegalizeTfTypesPass`. This pass assume that all quantized type can be safely removed, which means we have to rewriter all quantized ops before this pass. However, the pre-defined pass pipeline does not support register customized quantized op lowering patterns.